addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -342,6 +351,17 @@ class MessageList extends Component {
         }
     }
 
+    /**
+     * @private
+     * @param {Object} hint
+     */
+    _adjustFromChatWindowUnfolded(hint) {
+        if (this.threadViewer.threadCacheInitialScrollPosition !== undefined) {
+            this.el.scrollTop = this.threadViewer.threadCacheInitialScrollPosition;
+        }
+        this.threadViewer.markComponentHintProcessed(hint);
```

Comment:
This is a generic comment for all of them.

I think we might want to prevent all of those if `this.props.hasScrollAdjust` is false, similar to `_adjustFromChangeOfThreadCache` for example.

Then maybe we need a private method for this ("_adjustScrollFromModel" or similar), because those 3 or 4 are basically all copy/paste.

Reply:
there was something with hasScrollAdjust but @alexkuhn made me remove it

Reply:
Basically it was the same as in `adjustFromChangeOfThreadCache` :
```js
if (this.threadViewer.threadCacheInitialScrollPosition !== undefined) {
    if (this.props.hasScrollAdjust) {
        this.el.scrollTop = this.threadViewer.threadCacheInitialScrollPosition;
    }
}
```

Reply:
I suggested to remove because I thought that recovering scroll was not really an adjustment, but in the end it has been turned as an adjustment so ok for it being affected by this prop.